WordPress/openverse-frontend
# Image cell focus state improvements

## Fixes


Fixes #1004 by @zackkrida 

## Description



This PR makes several improvements to the ImageCell component:

- Adds a pink focus ring
- Unifies the hover/focus state that shows the CC license icons
- Renames to `VImageCell`. The existing `VImageCell` in the `VAllResults` grid is reamed to `VImageCellSquare`.
- Removes unnecessary custom styles
- Removes the unused `shouldContainImage` prop
- Gives the license icons the correct color in the production build
- Updates the gap between images in the VImageGrid to match the VAllResults grid

This PR also makes it quite evident that we should be able to unify the two image cell components, with an `aspect-ratio` prop for switching between square and aspect ratio appearances. The differences should be pretty straightforward to extract into a composable and some conditional logic in the template; I'll create a new issue for that as well.

Finally, this PR adds an offset to the BoxAudio focus ring, to match the ImageCell components.

## Screenshots

<img width="1277" alt="CleanShot 2022-02-25 at 10 47 37@2x" src="https://user-images.githubusercontent.com/6351754/155751273-8c29b482-3b04-4ebb-af71-15aa5143bcb1.png">

<img width="1243" alt="CleanShot 2022-02-25 at 10 47 48@2x" src="https://user-images.githubusercontent.com/6351754/155751299-79f5885c-512b-47f0-ba65-e7d4a46baee6.png">


## Testing Instructions


- `pnpm build && pnpm start`, then bavigate through the All results and image views via keyboard. Observe:
  - Correct focus states for  images and boxed audio
  - Correct icon color
- Look at related images and verify correct styles there too.


## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
